src/Core/src/Handlers/Border/BorderHandler.iOS.cs

```diff
@@ -51,8 +52,18 @@ static partial void UpdateContent(IBorderHandler handler)
 			if (handler.VirtualView.PresentedContent is IView content)
 			{
 				var platformContent = content.ToPlatform(handler.MauiContext);
-				platformContent.Tag = ContentView.ContentTag;
-				platformView.AddSubview(platformContent);
+
+				// If the content is a UIScrollView, we need a container to handle masks and clip shapes effectively.
+				if (platformContent is UIScrollView && content.Handler is not null)
+				{
+					content.Handler.HasContainer = true;
```

Comment:
I think with how HasContainer works this might be risky

Something might cause Container to reset back to false

![image](https://github.com/user-attachments/assets/cc77bad8-64c8-4c61-b9b5-9d608a423cca)

I would be curious at some point to explore a way for the parent to flag the child as needing a container but for now I think we'll just have to add a view here and then add the content to that view.